src/sql/workbench/api/common/extHostModelView.ts

```diff
@@ -1602,6 +1602,16 @@ class DeclarativeTableWrapper extends ComponentWrapper implements azdata.Declara
 		this._proxy.$doAction(this._handle, this._id, ModelViewAction.Filter, rowIndexes);
 	}
 
+	public get selectedRow(): number {
+		return this.properties['selectedRow'] ?? -1;
+	}
+
+	public set selectedRow(v: number) {
+		this.setProperty('selectedRow', v);
+		const emitter = this._emitterMap.get(ComponentEventType.onDidClick);
```

Comment:
onDidClick? Seems like making a new event here would be clearer (especially if we're allowing setting the row programmatically so a row being selected doesn't always mean it's clicked)

Reply:
Looking through the existing code it looks like this is probably because this is the event currently being fired (I assume it was chosen since previously it was only changed from clicking).

I guess then we can't really change it without risking affecting existing users...that's unfortunate though - we should definitely try to be more careful when choosing events in the future to try and not be too loose with picking the event for an action.